rt/src/scheduler/mod.rs

```diff
@@ -6,17 +6,17 @@ use std::thread::available_parallelism;
 
 #[cfg(target_os = "linux")]
 use {
-    libc::{cpu_set_t, sched_setaffinity, CPU_SET},
-    std::mem::{size_of, zeroed},
+    rustix::process::{sched_setaffinity, CpuSet, Pid},
+    std::mem::zeroed,
 };
 
 #[cfg(target_os = "linux")]
 pub(crate) fn pin_thread_to_core(core: usize) {
     unsafe {
-        let mut set: cpu_set_t = zeroed();
+        let mut set = zeroed();
 
-        CPU_SET(core, &mut set);
-        sched_setaffinity(0, size_of::<cpu_set_t>(), &set);
+        CpuSet::set(&mut set, core);
```

Comment:
I missed this before I think, but this isn't correct. `CpuSet::set` is an instance method (see [here](https://docs.rs/rustix/latest/rustix/process/struct.CpuSet.html#method.set)), and expects an initialised `self`. But what you're doing here is passing it zeroed-out memory. While that _may_ work, this is probably by coincidence, and if the memory is expected to be different in the future (i.e. not all zeroes), this will fail.

Or to put it differently: this line invokes undefined behaviour that happens to work by coincidence.

Looking at the Rustix API, we can do this instead:

```rust
let mut set = CpuSet::new();

set.set(core);
```